app/src/main/java/com/example/music_app/data/LoginStore.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.music_app.data
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+
+private const val NAME = "LoginStore"
+
+
+class LoginStore(private val context: Context?) {
```

Comment:
Please make this class a singleton and name it more general like `DataStoreManager`